develop/parachain-devs/runtime-development/FRAME/create-custom-pallet.md

```diff
@@ -0,0 +1,334 @@
+
+This step adds an event called `SomethingStored`, which is triggered when a user successfully stores a value in the pallet. The event records both the value and the account that performed the action.
+
+To define events, replace the `#[pallet::event]` line with the following code block:
+
+```rust
+#[pallet::event]
+#[pallet::generate_deposit(pub(super) fn deposit_event)]
+pub enum Event<T: Config> {
+    /// A user has successfully set a new value.
+    SomethingStored {
+        /// The new value set.
+        something: u32,
+        /// The account who set the new value.
+        who: T::AccountId,
+    },
+}
+```
+
+## Pallet Errors
+
+While events signal the successful completion of calls, errors indicate when and why a call has failed. It's essential to use informative names for errors to clearly communicate the cause of failure. Like events, error documentation is included in the node's metadata, so providing helpful descriptions is crucial.
+
+Errors are defined as an enum named `Error` with a generic type. Variants can have fields or be fieldless. Any field type specified in the error must implement the `scale_info::TypeInfo` trait, and the encoded size of each field should be as small as possible. Runtime errors can be up to 4 bytes in size, allowing the return of additional information when needed.
+
+This step defines two basic errors: one for handling cases where no value has been set, and another for managing arithmetic overflow.
+
+To define errors, replace the `#[pallet::error]` line with the following code block:
+
+```rust
+#[pallet::error]
+pub enum Error<T> {
+    /// The value retrieved was `None` as no value was previously set.
+    NoneValue,
+    /// There was an attempt to increment the value in storage over `u32::MAX`.
+    StorageOverflow,
+}
+```
+
+## Pallet Storage
+
+To store data in the pallet, the `#[pallet::storage]` macro is used. This macro allows the definition of abstract storage within the runtime and sets metadata for that storage. It can be applied multiple times to define different storage items. There are several types available for defining storage, which can be explored in the [Polkadot SDK documentation](https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/types/index.html){target=\_blank}.
+
+This step adds a simple storage item, `Something`, which stores a single `u32` value in the pallet's runtime storage
+
+To define storage, replace the #[pallet::storage] line with the following code block:
+
+```rust
+#[pallet::storage]
+pub type Something<T> = StorageValue<_, u32>;
+```
+
+## Pallet Calls
+
+Dispatchable functions enable users to interact with the pallet and trigger state changes. These functions are represented as "extrinsics," which are similar to transactions. They must return a `DispatchResult` and be annotated with both a weight and a call index.
```

Comment:
add the following reference https://paritytech.github.io/polkadot-sdk/master/frame_support/dispatch/type.DispatchResult.html for `DispatchResult`